nodes/client/client.js

```diff
@@ -5,30 +5,46 @@ module.exports = function (RED) {
     this.ozmapconnection = RED.nodes.getNode(config.ozmapconnection);
     this.status({});
     this.on('input', async (msg) => {
-      let ozmap = msg.ozmap || this.ozmapconnection.ozmap;
-      if (!ozmap.isConnected()) {
-        msg.payload = 'Ozmap not connected!';
-        this.status({ fill: 'red', shape: 'ring', text: 'disconnected' });
-        return this.send([null, msg]);
+      let ozmap = msg.ozmap || this.ozmapconnection?.ozmap;
+      if (!ozmap) {
+        msg.payload = 'Missing ozmap connection';
+        return [msg, null];
       }
+      this.status({});
+
       try {
-        if (msg.payload.query) {
-          const query = JSON.parse(msg.payload.query);
-          msg.payload = await ozmap.getClient().getAllByQuery(query);
-        } else if (msg.payload.filters) {
-          msg.payload = await ozmap.getClient().getAllByFilter(msg.payload.filters);
-        } else if (msg.payload.ids) {
-          msg.payload = await ozmap.getClient().getByIds(msg.payload.ids);
+        this.status({ fill: 'blue', shape: 'ring', text: 'running' });
+        const options = {
+          limit: config.limit,
+          page: config.page,
+          select: config.select,
+          filter: config.filter,
+          populate: config.populate,
+          sorter: config.sorter,
+        };
+        Object.assign(options, msg.payload);
```

Comment:
pq do assign como um passo separado? n aceita spread?
```suggestion
          sorter: config.sorter,
          ...msg.payload,
        };
```